server/docs/design/block-verification.md

```diff
@@ -0,0 +1,198 @@
+# Block Verification Design
+
+## Table of Contents
+
+1. [Purpose](#purpose)
+2. [Goals](#goals)
+3. [Terms](#terms)
+4. [Entities](#entities)
+5. [Design](#design)
+6. [Interfaces](#interfaces)
+7. [Enums](#enums)
+8. [Metrics](#metrics)
+9. [Exceptions](#exceptions)
+
+## Purpose
+
+The purpose of the Block Verification feature is to ensure that blocks received from consensus nodes are valid and have not been tampered with. This is achieved by re-calculating the block hash and verifying it against the signature provided by the consensus node.
+
+## Goals
+
+1) The block-node must re-create the block hash from the block items and verify that it matches the hash implied by the signature.
+2) If verification fails, the block should be considered invalid, and appropriate error-handling procedures must be triggered.
```

Comment:
Markdown standard is a `.` for ordered lists
```suggestion
1. The block-node must re-create the block hash from the block items and verify that it matches the hash implied by the signature.
1. If verification fails, the block should be considered invalid, and appropriate error-handling procedures must be triggered.
```

Reply:
✅ Done!